ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/util/HexStringComparator.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner.core.util;
+
+import java.util.Comparator;
+
+import org.apache.tuweni.bytes.Bytes;
+
+public class HexStringComparator implements Comparator<String> {
+
+  @Override
+  public int compare(String l, String r) {
+    final Bytes leftBytes = Bytes.fromHexString(l);
```

Comment:
it might not be a use case, but you can add additional conditions for null checks. if both l, r null, return true. if one of them null, return false?

Reply:
or raise NPE stating null values not allowed? (or it might already be raised fromHexString)

Reply:
thinking NPE is a good idea

Reply:
actually - the fromHexString performs said notnullcheck, but will raise IllegalArgumentException